docs/pages/database-access/auto-user-provisioning/database-object-permissions.mdx

```diff
@@ -0,0 +1,191 @@
+---
+title: Database object permissions
+description: Understand the concepts behind database object permissions managed by Teleport.
+---
+
+## Overview
+
+Database object permissions is a Teleport that enables the configuration of access levels for individual users to specific database objects based on their required level of access.
+
+Database objects are labeled based on import rules, and these labels correspond to specific permissions configured within user roles.
+
+Permissions are granted for the duration of a connection and are automatically revoked upon its closure.
+
+## Database object import rules
+
+A database object import rule in Teleport is a resource that defines the labels to be applied to database objects imported into Teleport. If a specific object does not match any of the rules, it will not be imported.
+
+By default, if no import rules are present, Teleport will automatically create the `import_all_objects` rule on startup:
+
+```yaml
+kind: db_object_import_rule
+metadata:
+  name: import_all_objects
+spec:
+  # Priority determines how important the rule is.
+  # In case of conflicts, when the same label is applied by two rules, 
+  # the label applied by rule with higher priority wins.
+  priority: 0
+  # database_labels is a filter specifiying which databases are in scope of this rule.
+  database_labels:
+  - name: '*'
+    values:
+    - '*'
+  # Each mapping, if matched, introduces a set of labels applied to database object.
+  # Database objects without labels are not imported.
+  mappings:
+  - add_labels:
+      database: '{{obj.database}}'
+      object_kind: '{{obj.object_kind}}'
+      name: '{{obj.name}}'
+      protocol: '{{obj.protocol}}'
+      schema: '{{obj.schema}}'
+      database_service_name: '{{obj.database_service_name}}'
+    match:
+      procedure_names:
+      - '*'
+      table_names:
+      - '*'
+      view_names:
+      - '*'
+  # Additional mappings can be added here.
+  # - add_labels: ...
+version: v1
+```
+
+This rule will import all objects and label them by their inherent properties using the template syntax.
+
+Feel free to customize this rule to meet your specific requirements or add more rules. For instance, consider the following rule designed to designate particular tables as accessible to developers, either in a read-only or read-write capacity.
+
+```yaml
+kind: db_object_import_rule
+metadata:
+  name: developer_read_write
+spec:
+  priority: 100
+  database_labels:
+  # Affect only `staging` and `dev` environments.
+  - name: 'env'
+    values:
+    - 'staging'
+    - 'dev'
+  mappings:
+  # Apply `read_write` to all tables in given schemas.
+  - add_labels:
+      developer_permissions: read_write
+    match:
+      table_names:
+      - '*'
+    scope:
+      schema_names:
+      - 'application'
+      - 'data_import'
```

Comment:
nit: i think we should have an example of `database_names` (it could be commented out) and add some comments on the default behavior when not specified.

Reply:
Done, let me know what you think.